File: model/src/main/java/net/artux/pda/model/items/ItemModel.java
package net.artux.pda.model.items;

import java.io.Serializable;
import java.util.UUID;

import lombok.Data;

@Data
public class ItemModel implements Serializable {

    protected UUID id;
    protected ItemType type;
    protected String icon;
    protected String title;
    protected int baseId;
    protected float weight;
    protected int price;
    protected int quantity;

}

File: model/src/main/java/net/artux/pda/model/items/ItemModel.kt
package net.artux.pda.model.items

import java.io.Serializable
import java.util.UUID

open class ItemModel : Serializable {
    var id: UUID? = UUID.randomUUID()
    var type: ItemType = ItemType.ITEM
    var icon: String? = null
    var title: String? = null

    @JvmField
    var baseId = 0
    var weight = 0f
    var price = 0
    var quantity = 0
}